File: apps/frontend/lib/core/models/episode.dart
class Episode {
  final String id;
  final String projectId;
  final String title;
  final String? description;
  final int episodeNumber;
  final String status;
  final DateTime createdAt;
  final DateTime updatedAt;

  Episode({
    required this.id,
    required this.projectId,
    required this.title,
    this.description,
    required this.episodeNumber,
    required this.status,
    required this.createdAt,
    required this.updatedAt,
  });

  factory Episode.fromJson(Map<String, dynamic> json) {
    return Episode(
      id: json['id'],
      projectId: json['project_id'],
      title: json['title'],
      description: json['description'],
      episodeNumber: json['episode_number'],
      status: json['status'],
      createdAt: DateTime.parse(json['created_at']),
      updatedAt: DateTime.parse(json['updated_at']),
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'project_id': projectId,
      'title': title,
      'description': description,
      'episode_number': episodeNumber,
      'status': status,
    };
  }
}

File: apps/backend/app/models/episode.py
import uuid
from datetime import datetime, timezone
from sqlalchemy import Column, String, Integer, Text, DateTime, ForeignKey
from sqlalchemy.dialects.postgresql import UUID
from sqlalchemy.orm import relationship

from app.models.project import Base


class Episode(Base):
    __tablename__ = "episodes"

    id = Column(UUID(as_uuid=True), primary_key=True, default=uuid.uuid4)
    project_id = Column(UUID(as_uuid=True), ForeignKey("projects.id", ondelete="CASCADE"), nullable=False)
    title = Column(String(255), nullable=False)
    description = Column(Text, nullable=True)
    episode_number = Column(Integer, nullable=False)
    status = Column(String(50), default="draft")
    created_at = Column(DateTime, default=lambda: datetime.now(timezone.utc))
    updated_at = Column(DateTime, default=lambda: datetime.now(timezone.utc), onupdate=lambda: datetime.now(timezone.utc))

    project = relationship("Project", back_populates="episodes")
    story = relationship("Story", back_populates="episode", uselist=False, cascade="all, delete-orphan")

    def __repr__(self):
        return f"<Episode {self.episode_number}: {self.title}>"

